lib/developer_portal/app/controllers/developer_portal/base_controller.rb

```diff
@@ -25,4 +25,11 @@ def finish_signup_for_paid_plan
 
     redirect_to edit_payment_details_path
   end
+
+  def filter_readonly_params(params, resource_class)
```

Comment:
One question came to mind. What if user submits parameters that don't exist as custom fields? We don't filter out such? Can they update something else that is not supposed to be updated?

Reply:
I think this is prevented by the assignment method:
https://github.com/3scale/porta/blob/1cb77853cb82d3b66d3012137d2b621bb3e24643/app/lib/fields/extra_fields.rb#L43-L55

So I guess whatever non-existing fields will be simply ignored.

Reply:
Sounds good bu I see here:
https://github.com/3scale/porta/blob/767e100b92dea8ca71e5d24ab05155632b37e04a/lib/developer_portal/app/controllers/developer_portal/admin/accounts_controller.rb#L24

Which tracking it down, it appears not using this mechanism.

I think it would be safer for the controller to filter not only read-only but also non-defined fields. Maybe rename the method to `only_writable_extra_fields` or something along the lines.

Reply:
For `Account` and `Cinstance`, there's a black list of protected attributes that can't be set in mass assignment, so all other attrs can be set, unless they are stopped by strong params or any other mechanism.

For `User`, there's a white list of protected attributes, so only attrs in that list can be mass-assigned.

That's out of scope for this PR I think. This PR doesn't improve or worsens the situation.

Reply:
Just it makes sense to me if we filter the paremeters, then just do it completely. Other layers should be an additional like of protection. If there are no negative effects of updating this method to exclude anything not writable, including non-existing, I'd say it is easy to do now.

If others are ok with current state, that's also acceptable to me. Just not ideal.

Reply:
At the beginning I thought on using Strong parameters to set up a white list of permitted parameters, but I'm not sure of this because these controllers face developer portals and we don't know what are clients really sending to here, so it would potentially break some portals. That's why I think it's better to investigate and do this in another PR

Reply:
escapist